### PR TITLE
queue subscription route update set

### PR DESCRIPTION
**Pull request by Galor-sw**

### CODE REVIEW

**Review by itay45977 — APPROVED** [review]

File: RMQ/reciverQueueMessage.js [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
-                const jsonMessage = (JSON.parse(msg.content.toString()));
-                logger.info(`message from IAM - set free plan to: ${jsonMessage.accountId}`);
-            }, {noAck: true});
-        })
-    })
-
+  logger.info('listening to IAM queues');
+  amqp.connect(IAMMessageCreateFreePlan, (err, conn) => {
+    conn.createChannel((err, ch) => {
+      const q = 'CloudAMQP';
```
Comment (dorhaimovich) [inline RMQ/reciverQueueMessage.js:13]: named variable

Comment (dorhaimovich) [reply RMQ/reciverQueueMessage.js:13]: better to change it

File: cronJob/cronJob.js [hunk 2]
```diff
@@ -1,57 +1,54 @@
...
-            sendSubscriptionToIAM(subscription.accountId, subscription.plan.credits, subscription.plan.seats, subscription.plan.features);
-        }
-    })
-}
+  const nextMonth = moment().add(1, 'M').format('YYYY-MM-DD HH:mm:ss');
+  let updatedSubscription = new Subscription();
+  updatedSubscription = subscription;
+  updatedSubscription.next_date = nextMonth;
+
+  Subscription.findByIdAndUpdate(subscription._id, updatedSubscription, { new: true }, (err) => {
```
Comment (dorhaimovich) [inline cronJob/cronJob.js:16]: call the Repo of the subscription instead

File: cronJob/cronJob.js [hunk 3]
```diff
@@ -1,57 +1,54 @@
...
-                        updateSubscription(subscription);
-                    }
-
-                });
-            }
+  let subscriptions;
+  const format = 'YYYY-MM-DD';
+  cron.schedule('00 02 * * *', () => {
+    // get all free subscriptions
+    axios.get('http://localhost:5000/subscription/getAllSubscriptionsByPlanName/Free')
```
Comment (dorhaimovich) [inline cronJob/cronJob.js:31]: path should be retrieve from env

File: cronJob/cronJob.js [hunk 4]
```diff
@@ -1,57 +1,54 @@
...
+      .then(subscriptionsResult => {
+        subscriptions = subscriptionsResult.data;
+      })
+      .catch(err => {
```
Comment (dorhaimovich) [inline cronJob/cronJob.js:35]: send err.message

**Review by dorhaimovich — APPROVED** [review]

**Pull request merged**